### PR TITLE
Log entry filtering

### PR DESCRIPTION
**Pull request by benni043**
this pr adds

1. filtering by ip
2. filtering by date
3. filtering by text
4. filtering by regex
5. filtering by classification

### CODE REVIEW

**Review by xFizzar — APPROVED** [review]
looks good to me

File: src/entry/logEntry.service.ts [hunk 1]
```diff
@@ -31,19 +31,40 @@ export class LogEntryService {
...
+      let date;
+
+      // filter
+      if (logEntryData.filters !== undefined) {
+        ip = logEntryData.filters.ip;
+        text = logEntryData.filters.text;
+        regex = logEntryData.filters.regex;
+        classification = logEntryData.filters.classification;
+        date = logEntryData.filters.date;
+      }
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.service.ts]: create type Filters with those values as instance variables. No need to assign them to individual varaibles

File: src/entry/logEntry.service.ts [hunk 2]
```diff
@@ -31,19 +31,40 @@ export class LogEntryService {
...
         sessionID,
         files,
         from,
         count,
         sortingOrderDESC === undefined ? false : sortingOrderDESC,
+        ip,
+        text,
+        regex,
+        classification,
+        date,
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.service.ts]:
```suggestion
        filters
```

File: src/entry/logEntry.store.ts [hunk 3]
```diff
@@ -19,21 +19,65 @@ export class LogEntryStore {
...
-            SELECT *
+      const queryParams: any[] = [sessionID, files];
+      let queryString = `
+      SELECT *
             FROM loggaroo.log_entry
               WHERE session_id = $1
                 AND file_name = ANY($2)
+                `;
+
+      if (ip) {
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.store.ts]:
```suggestion
      if (filters.ip) {
```

File: src/entry/logEntry.store.ts [hunk 4]
```diff
@@ -19,21 +19,65 @@ export class LogEntryStore {
...
+        queryParams.push(ip);
+        queryString += "AND service_ip = $" + queryParams.length;
+      }
+
+      if (text) {
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.store.ts]:
```suggestion
      if (filters.text) {
```

File: src/entry/logEntry.store.ts [hunk 5]
```diff
@@ -19,21 +19,65 @@ export class LogEntryStore {
...
+        queryParams.push(text);
+        queryString += "AND content = $" + queryParams.length;
+      }
+
+      if (regex) {
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.store.ts]:
```suggestion
      if (filters.regex) {
```

File: src/entry/logEntry.store.ts [hunk 6]
```diff
@@ -19,21 +19,65 @@ export class LogEntryStore {
...
+        queryParams.push(regex);
+        queryString += "AND content ~ $" + queryParams.length;
+      }
+
+      if (classification) {
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.store.ts]:
```suggestion
      if (filters.classification) {
```

File: src/entry/logEntry.store.ts [hunk 7]
```diff
@@ -19,21 +19,65 @@ export class LogEntryStore {
...
+        queryParams.push(classification);
+        queryString += "AND service_ip = $" + queryParams.length;
+      }
+
+      if (date) {
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.store.ts]:
```suggestion
      if (filters.date) {
```

File: src/entry/logEntry.store.ts [hunk 8]
```diff
@@ -19,21 +19,65 @@ export class LogEntryStore {
...
+        queryParams.push(date.from);
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.store.ts]: make it so that from can be null

File: src/entry/logEntry.store.ts [hunk 9]
```diff
@@ -19,21 +19,65 @@ export class LogEntryStore {
...
+
+        if (date.to) queryParams.push(date.to);
+        else queryParams.push(new Date(Date.now()).toISOString().split("T")[0]);
+
+        queryString += `AND creation_date BETWEEN $${
+          queryParams.length - 1
+        } AND $${queryParams.length}`;
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.store.ts]: If one is missing make it only to check if the date is < or >

File: src/entry/logEntry.store.ts [hunk 10]
```diff
@@ -19,21 +19,65 @@ export class LogEntryStore {
     from: number,
     count: number,
     sortingOrderDESC: boolean,
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.store.ts]:
```suggestion
    order: "DESC" | "ASC",
```

File: src/entry/logEntry.ts [hunk 11]
```diff
@@ -22,7 +22,21 @@ export type LogEntryRequestData = {
...
   sortingOrderDESC: boolean | undefined;
+
+  filters: {
+    date: RangeDate | undefined;
+    ip: string | undefined;
+    text: string | undefined;
+    regex: string | undefined;
+    classification: string | undefined;
+  };
+};
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.ts]: make this into a "Filter" type

File: src/entry/logEntry.store.ts [hunk 12]
```diff
@@ -19,21 +19,65 @@ export class LogEntryStore {
...
+    ip: string | undefined,
+    text: string | undefined,
+    regex: string | undefined,
+    classification: string | undefined,
+    date: RangeDate | undefined,
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.store.ts]:
```suggestion
    filters: Filters
```

File: src/entry/logEntry.store.ts [hunk 13]
```diff
@@ -19,21 +19,65 @@ export class LogEntryStore {
...
   ): Promise<LogEntry[] | LogEntryRequestError> {
     try {
       const order = sortingOrderDESC ? "DESC" : "ASC";
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.store.ts]:
```suggestion

```

File: src/entry/logEntry.ts (hunk 11 above)
Comment (JonasPfeifer05) [reply src/entry/logEntry.ts]: should also have boolean "isRegex" and one field to pass either the text or regex

File: src/entry/logEntry.ts [hunk 14]
```diff
@@ -22,7 +22,21 @@ export type LogEntryRequestData = {
...
+
+export type RangeDate = {
+  from: Date;
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.ts]:
```suggestion
  from: Date | null;
```

**Review by JonasPfeifer05 — CHANGES_REQUESTED** [review]
please change

File: src/entry/logEntry.service.ts [hunk 15]
```diff
@@ -22,29 +22,24 @@ export class LogEntryService {
     logEntryData: LogEntryRequestData,
   ): Promise<LogEntry[] | LogEntryRequestError> {
     try {
-      const files: string[] = logEntryData.files;
       const from: number = logEntryData.from;
       const count: number = logEntryData.count;
-      const sortingOrderDESC = logEntryData.sortingOrderDESC;
+      const sortingOrderDESC = logEntryData.order;
 
       if (from < 0 || count < 0) {
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.service.ts]:
```suggestion
      if (from < 0 || count <= 0) {
```

File: src/entry/logEntry.service.ts [hunk 16]
```diff
@@ -22,29 +22,24 @@ export class LogEntryService {
...
         return LogEntryRequestError.wrongBodyData;
       }
 
+      // order
       if (
-        typeof sortingOrderDESC !== "boolean" &&
+        sortingOrderDESC !== "ASC" &&
+        sortingOrderDESC !== "DESC" &&
         typeof sortingOrderDESC !== "undefined"
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.service.ts]:
```suggestion
        typeof sortingOrderDESC !== "undefined" &&
        sortingOrderDESC !== "ASC" &&
        sortingOrderDESC !== "DESC"
```

File: src/entry/logEntry.store.ts [hunk 17]
```diff
@@ -15,25 +19,73 @@ export class LogEntryStore {
...
+        if (logEntryRequestData.filters.regex) {
+          queryString += "AND content ~ $" + queryParams.length;
+        } else {
+          queryString += "AND content = $" + queryParams.length;
+        }
+      }
+
+      if (logEntryRequestData.filters?.classification) {
+        queryParams.push(logEntryRequestData.filters.classification);
+        queryString += "AND service_ip = $" + queryParams.length;
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.store.ts]:
```suggestion
      if (logEntryRequestData.filters?.classification) {
        queryParams.push(logEntryRequestData.filters.classification);
        queryString += "AND classification = $" + queryParams.length;
```

**Review by JonasPfeifer05 — CHANGES_REQUESTED** [review]

**Review by JonasPfeifer05 — APPROVED** [review]
ok

**Pull request merged**